internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -815,27 +837,33 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 			return diags
 		}
 		clusterChangeDetect := new(admin20240530.AdvancedClusterDescription)
+		var waitOnUpdate bool
 		if !reflect.DeepEqual(req, clusterChangeDetect) {
 			if err := CheckRegionConfigsPriorityOrderOld(req.GetReplicationSpecs()); err != nil {
 				return diag.FromErr(err)
 			}
 			if _, _, err := connV220240530.ClustersApi.UpdateCluster(ctx, projectID, clusterName, req).Execute(); err != nil {
 				return diag.FromErr(fmt.Errorf(errorUpdate, clusterName, err))
 			}
-			if err := waitForUpdateToFinish(ctx, connV2, projectID, clusterName, timeout); err != nil {
-				return diag.FromErr(fmt.Errorf(errorUpdate, clusterName, err))
-			}
-		} else if d.HasChange("replica_set_scaling_strategy") || d.HasChange("redact_client_log_data") {
+			waitOnUpdate = true
+		}
+		if d.HasChange("replica_set_scaling_strategy") || d.HasChange("redact_client_log_data") || d.HasChange("config_server_management_mode") {
```

Comment:
fix: supports updating both old cluster attributes and new attributes when using old SDK